tag-system/src/main/java/org/togetherjava/tjbot/tag/TagSystem.java

```diff
@@ -0,0 +1,66 @@
+package org.togetherjava.tjbot.tag;
+
+import org.jooq.Result;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.Tags;
+import org.togetherjava.tjbot.db.generated.tables.records.TagsRecord;
+
+import java.util.*;
+
+public class TagSystem {
+    private final Database database;
+
+    public TagSystem(Database database) {
+        this.database = database;
+    }
+
+    public boolean exists(String tag) {
+        return database.readTransaction(ctx -> {
+            return Optional
+                .ofNullable(ctx.selectFrom(Tags.TAGS).where(Tags.TAGS.ID.eq(tag)).fetchOne())
+                .isPresent();
+        });
+    }
+
+    public void delete(String tag) {
+        database.writeTransaction(ctx -> {
+            ctx.deleteFrom(Tags.TAGS).where(Tags.TAGS.ID.eq(tag)).execute();
+        });
+    }
+
+    public void put(String tag, String content) {
+        database.writeTransaction(ctx -> {
+            TagsRecord tagsRecord = ctx.newRecord(Tags.TAGS).setId(tag).setContent(content);
+
+            if (tagsRecord.update() == 0) {
+                tagsRecord.insert();
+            }
+        });
+    }
+
+    public String get(String tag) {
+        return database.readTransaction(ctx -> {
+            return Optional
+                .ofNullable(ctx.selectFrom(Tags.TAGS).where(Tags.TAGS.ID.eq(tag)).fetchOne())
+                .map(TagsRecord::getContent)
+                .orElse(null);
```

Comment:
Returning a null Optional is weird. Should we return "Tag not found" or something similar?

Reply:
No, gigo (garbage in garbage out).

Reply:
why?

Reply:
I'd rather get a NPE and fail-hard instead of returning something not useful when the input's clearly wrong.

Reply:
An optional and a NPE?
Why return an optional if you want a NPE

Reply:
I'm not returning an optional.

Reply:
Didn't notice the `database.readTransaction` cause I was reading from the conversation tab

Reply:
Sorry but I do not get this. Why do you return `null` and _abuse a `Optional` just for its API here? That is kinda misleading and usually considered a nono.

If you consider it an error, throw an exception. If not, return an `Optional`. Prefer not using `null` as return value please.
And in particular, do not abuse `Optional` just for its "nice API". If you do not want to return an `Optional`, then use normal if-else please. Then again, why not just return `Optional` out of this method?